sys/net/application_layer/dns/dns.c

```diff
@@ -157,21 +157,21 @@ static int _parse_dns_reply(uint8_t *buf, size_t len, void* addr_out, int family
             return -EBADMSG;
         }
         bufpos += RR_RDLENGTH_LENGTH;
-        if ((bufpos + addrlen) >= buflim) {
+        if ((bufpos + addrlen) > buflim) {
             return -EBADMSG;
         }
 
         memcpy(addr_out, bufpos, addrlen);
-        return addrlen;
+        return 0;
     }
 
     return -1;
 }
 
+static uint8_t reply_buf[512];
```

Comment:
The replies I got were 56 bytes max - do we really need such a huge buffer?

Reply:
Might be even better to make it compile-time configurable with default of 64?

Reply:
I've decided for 128 bytes (phoronix.com gives me a 86 byte reply), but to re-use the reply buffer for the request too as we never need both at the same time.